### PR TITLE
improve tests

### PR DESCRIPTION
**Pull request by khassel**
use es6 syntax in all tests, split weather tests, remove callbacks

### CODE REVIEW

**Comment by khassel** [comment]
@MichMich I like to remind you to change project settings to "squashing" as described [here](https://github.com/MichMich/MagicMirror/issues/2916#issuecomment-1259881088) **before** merging this, thanks.

**Comment by rejas** [comment]
Thanks for your work. One favor I have to ask: Could you split the weather tests into two files, weather_current_spec and weather_forecast_spec? Would help me a lot when cleaning them up in my weather refactor later.

File: tests/e2e/modules/clock_spec.js [hunk 1]
```diff
@@ -7,8 +7,7 @@ describe("Clock module", () => {
 	});
 
 	const testMatch = (done, element, regex) => {
```
Comment (rejas) [inline tests/e2e/modules/clock_spec.js:9]: general question: cant the tests themself not be written async too? something like

const testMatch = async (elemtn, regex) .... ?

Comment (rejas) [reply tests/e2e/modules/clock_spec.js:9]: currently it looks like a wild mix of callback and async....

Comment (khassel) [reply tests/e2e/modules/clock_spec.js:9]: can you give me a more detailed example how `testMatch` should look like and how it should be called?

Comment (khassel) [reply tests/e2e/modules/clock_spec.js:9]: think I got it, will deliver new commit.

**Review by rejas — APPROVED** [review]
Awesome! SO much cleaner now!

**Review by rejas — APPROVED** [review]
Approved and eager to have it merged @MichMich  :-)

**Comment by MichMich** [comment]
🎉

**Pull request merged**